app/src/main/java/com/github/se/signify/ui/Utils.kt

```diff
@@ -824,6 +827,30 @@ fun CameraPlaceholder(handLandMarkViewModel: HandLandMarkViewModel, modifier: Mo
         }
   }
 }
+/**
+ * Updates the application's language dynamically by changing the locale.
+ *
+ * This function sets the application's locale to the specified language code and updates the
+ * configuration for the application's resources to reflect the language change.
+ *
+ * @param context The context of the application or activity. This is required to access the
+ *   application's resources and configuration.
+ * @param languageCode The language code for the desired language (e.g., "en" for English, "fr" for
+ *   French). This should be a valid ISO 639-1 language code.
+ *
+ * Usage Example:
+ * ```
+ * updateLanguage(context, "fr") // Switches the application language to French
+ * updateLanguage(context, "en") // Switches the application language to English
+ * *
+ */
+fun updateLanguage(context: Context, languageCode: String) {
+  val locale = Locale(languageCode)
+  Locale.setDefault(locale)
+  val config = Configuration()
+  config.locale = locale
+  context.resources.updateConfiguration(config, context.resources.displayMetrics)
+}
```

Comment:
The updateLanguage function currently assumes the languageCode provided is valid. I think adding validation for the language code (e.g., checking against a list of supported locales) would make it more robust.
```suggestion
fun updateLanguage(context: Context, languageCode: String) {
if (!listOf("en", "fr").contains(languageCode)) {
    throw IllegalArgumentException("Unsupported language code: $languageCode")
}
  val locale = Locale(languageCode)
  Locale.setDefault(locale)
  val config = Configuration()
  config.locale = locale
  context.resources.updateConfiguration(config, context.resources.displayMetrics)
}
```

Reply:
You are absolutly right, thanks for pointing that (Nice catch ! ).
And thank you for giving the code it's easy to integrate !